File: 1021_remove_outermost_parentheses/removeOuterParentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        res = []
        op = 0

        for c in s:
            if c == "(" and op > 0:
                res.append(c)
            if c == ")" and op > 1:
                res.append(c)
            if c == "(":
                op += 1
            else:
                op -= 1

        return "".join(res)

File: 1021_remove_outermost_parentheses/remove_outer_parentheses.rs
impl Solution {
    pub fn remove_outer_parentheses(s: String) -> String {
        let mut res = String::new();
        let mut level = 0;

        for c in s.chars() {
            if c == '(' {
                if level > 0 {
                    res.push(c);
                }

                level += 1;
            } else {
                if level > 1 {
                    res.push(c);
                }

                level -= 1;
            }
        }
        
        res
    }
}